icu4c/source/test/cintltst/cloctst.c

```diff
@@ -2508,6 +2508,9 @@ static void TestCanonicalization21749StackUseAfterScope(void)
                 input, u_errorName(status));
         return;
     }
+
+    status = U_ZERO_ERROR;
+    uloc_canonicalize("ti-defaultgR-lS-z-UK-0P", buffer, UPRV_LENGTHOF(buffer), &status);
```

Comment:
```suggestion

    // ICU-22475 test that we don't free an internal buffer twice.
    status = U_ZERO_ERROR;
    uloc_canonicalize("ti-defaultgR-lS-z-UK-0P", buffer, UPRV_LENGTHOF(buffer), &status);
```
... to help explain why this is here and why we don't check for the results of the function call.